Core/Resource/Model.cs

```diff
@@ -442,51 +444,86 @@ private static void RegisterType(Type type, string typeName, Func<InputValue> de
+                                                     jsonInfo.Object = symbol;
+                                                 }
+                                             }
+                                             else
+                                             {
+                                                 Log.Warning($"Failed to load symbol {jsonInfo.Guid}");
+                                             }
+
+                                             return (success, symbol);
+                                         })
+                                 .ToList(); // execute and bring back to main thread
+
+            foreach (var j in _symbolJsonTokens )
             {
-                ReadSymbolFromFile(symbolFilePath);
+                var jsonResult = j.Value;
+                if (jsonResult.ObjectWasSet)
+                {
+                    SymbolRegistry.Entries.TryAdd(jsonResult.Object.Id, jsonResult.Object);
+                }
             }
 
             // check if there are symbols without a file, if yes add these
             var instanceTypes = (from type in OperatorsAssembly.ExportedTypes
-                                 where type.IsSubclassOf(typeof(Instance))
+                                 where type.IsSubclassOf(InstanceType)
                                  where !type.IsGenericType
-                                 select type).ToList();
+                                 select type).ToHashSet();
 
             foreach (var symbol in SymbolRegistry.Entries.Values)
             {
-                for (int i = instanceTypes.Count - 1; i >= 0; i--)
-                {
-                    if (symbol.InstanceType == instanceTypes[i])
-                    {
-                        instanceTypes.RemoveAt(i);
-                        break;
-                    }
-                }
+                instanceTypes.Remove(symbol.InstanceType);
             }
 
             foreach (var newType in instanceTypes)
             {
-                string @namespace = _innerNamespace.Replace(newType.Namespace ?? string.Empty, "").ToLower();
-
-                string idFromNamespace = _idFromNamespace.Match(newType.Namespace ?? string.Empty).Value.Replace('_', '-');
-                Debug.Assert(!string.IsNullOrEmpty(idFromNamespace));
+                var typeNamespace = newType.Namespace;
+                if (string.IsNullOrWhiteSpace(typeNamespace))
+                {
+                    Log.Error($"Null or empty namespace of type {newType.Name}");
+                    continue;
+                } 
+                
+                var @namespace = _innerNamespace.Replace(typeNamespace, string.Empty).ToLowerInvariant();
+                var idFromNamespace = _idFromNamespace.Match(typeNamespace).Value;
+                idFromNamespace = ReplaceCharacter(idFromNamespace, '_', '-');
+                
+                Debug.Assert(!string.IsNullOrWhiteSpace(idFromNamespace));
```

Comment:
I optimized this to reduce `string` object allocations as much as possible using the `ReplaceCharacter` function.

Reply:
oh yeah your gut is absolutely correct and there's no noticeable difference to startup time and i have no problem just yeeting it